.github/workflows/install-sfdx-package.yml

```
@@ -244,3 +244,21 @@ jobs:
 
       - name: install package
         run: sf package install --package "${package}" --installation-key "${installation_key_decrypted}" --apex-compile package --upgrade-type Mixed --wait 15
+
+      - name: check out code
+        uses: actions/checkout@v4
+
+      - name: find post-deploy scripts
+        id: find-post-deploy-scripts
+        run: |
+          mkdir -p deploy/hooks/post-deploy
+          ls deploy/hooks/post-deploy/*.sh > post_deploy_scripts.txt || echo "No post-deploy scripts found" > post_deploy_scripts.txt
```

This would write the text "No post-deploy scripts found" to the `post_deploy_scripts.txt` file, which would then instruct your next step to try and execute the command `No post-deploy scripts found` with bash (which would most likely fail with a `No: command not found` error).

So I'd leave out the redirection and just echo this to stdout. You do need to handle the non-zero exit code that `ls` gives you when it doesn't find matching files, otherwise your whole GitHub Actions workflow job would fail so it's good to do the `|| echo` like you did. I'd also redirect stderr to `/dev/null` so you don't get `ls`'s error message when no files are found, since you're already printing out your own message in that case.

```suggestion
          mkdir -p deploy/hooks/post-deploy
          ls deploy/hooks/post-deploy/*.sh >post_deploy_scripts.txt 2>/dev/null || echo "No post-deploy scripts found"
```